File: 0581-shortest-unsorted-continuous-subarray/0581-shortest-unsorted-continuous-subarray.java
class Solution {
    public int findUnsortedSubarray(int[] nums) {
        int n=nums.length;
        int []sorted=nums.clone();
        Arrays.sort(sorted);
        int start=0;
        int end=n-1;
        while(start<n && sorted[start]==nums[start])
        start++;
        while(end>start && sorted[end]==nums[end])
        end--;
        return end-start+1;
    }
}

File: 0581-shortest-unsorted-continuous-subarray/0581-shortest-unsorted-continuous-subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        vector<int> v=nums;
        sort(v.begin(),v.end());
        int c=0;
        for(int i=0;i<v.size();i++)
        {
            if(v[i]!=nums[i])
            {
                c=i;
                break;
            }
        }
        for(int i=v.size()-1;i>=0;i--)
        {
            if(v[i]!=nums[i])
            {
                c=i-c+1;
                break;
            }
        }
        return c;
        
    }
};